include/vsg/platform/win32/Win32_Window.h

```diff
@@ -119,16 +119,40 @@ namespace vsgWin32
         return static_cast<vsg::ButtonMask>(mask);
     }
 
-    inline uint32_t getButtonDownEventDetail(UINT buttonMsg)
+    inline uint32_t getButtonDownEventDetail(UINT buttonMsg, WORD wParamHi)
     {
-        return buttonMsg == WM_LBUTTONDOWN ? 1 : (buttonMsg == WM_MBUTTONDOWN ? 2 : buttonMsg == WM_RBUTTONDOWN ? 3
-                                                                                                                : (buttonMsg == WM_XBUTTONDOWN ? 4 : 0)); // need to determine x1, x2
+        switch (buttonMsg)
+        {
+        case WM_LBUTTONDOWN: return 1;
+        case WM_MBUTTONDOWN: return 2;
+        case WM_RBUTTONDOWN: return 3;
+        case WM_XBUTTONDOWN:
+            if (wParamHi == XBUTTON1)
+                return 4;
+            else if (wParamHi == XBUTTON2)
+                return 5;
+            [[fallthrough]];
```

Comment:
return 0; is easier to read than [[fallthrough]] as it's more explicit about how this condition is meant to be treated, it's also less code.  Using a compiler hint to quieten a compile warning when just writing the code more explicitly to what you intend to happen is better.